test/performance/odp_packet_gen.c

```diff
@@ -965,6 +1142,107 @@ static int set_num_cpu(test_global_t *global)
 	return 0;
 }
 
+static int check_lso_capa(char *name, odp_pktio_capability_t *pktio_capa,
+			  test_options_t *opt, odp_pool_t pool)
+{
+	odp_packet_t pkt;
+	uint32_t max_segments;
+	uint32_t pkt_payload;
+	const uint32_t max_pkt_len = opt->use_rand_pkt_len ?
+				opt->rand_pkt_len_max : opt->pkt_len;
+
+	if (pktio_capa->lso.max_profiles < opt->num_pktio ||
+	    pktio_capa->lso.max_profiles_per_pktio == 0) {
+		ODPH_ERR("Error (%s): Not enough LSO profiles (max %" PRIu32 ")\n", name,
+			 ODPH_MIN(pktio_capa->lso.max_profiles,
+				  pktio_capa->lso.max_profiles_per_pktio));
+		return -1;
+	}
+
+	if (opt->lso.param.lso_proto == ODP_LSO_PROTO_TCP_IPV4 &&
+	    !pktio_capa->lso.proto.tcp_ipv4) {
+		ODPH_ERR("Error (%s): ODP_LSO_PROTO_TCP_IPV4 not supported\n", name);
+		return -1;
+	} else if (opt->lso.param.lso_proto == ODP_LSO_PROTO_IPV4 &&
+		   !pktio_capa->lso.proto.ipv4) {
+		ODPH_ERR("Error (%s): ODP_LSO_PROTO_IPV4 not supported\n", name);
+		return -1;
+	} else if (opt->lso.param.lso_proto == ODP_LSO_PROTO_CUSTOM &&
+		   !pktio_capa->lso.proto.custom) {
+		ODPH_ERR("Error (%s): ODP_LSO_PROTO_CUSTOM not supported\n", name);
+		return -1;
+	}
+
+	/* Check number of segments in a max length packet */
+	pkt = odp_packet_alloc(pool, max_pkt_len);
+	if (pkt == ODP_PACKET_INVALID) {
+		ODPH_ERR("Error (%s): Allocating test packet failed\n", name);
+		return -1;
+	}
+
+	max_segments = odp_packet_num_segs(pkt);
+	odp_packet_free(pkt);
```

Comment:
Yes, this is not an optimal solution but looks like there is none. I'd prefer this instead of capability check, since the capability check could produce "false failures" which would prevent using LSO, event though packets wouldn't have too many segments.

Reply:
OK, it's probably good enough this way. I guess we also lose the packet to control thread pool cache (?) but with potentially 100s to 1000s of packets in a pool that's a non-issue. Resolving.